fl4health/clients/moon_client.py

```diff
@@ -135,13 +146,22 @@ def compute_loss_and_additional_losses(
             "loss": loss,
         }
 
-        if self.contrastive_weight:
-            contrastive_loss = self.get_contrastive_loss(
-                features["features"], features["global_features"], features["old_features"]
+        if self.contrastive_weight and "old_features" in features:
+            contrastive_loss = self.contrastive_loss(
+                features["features"], features["global_features"].unsqueeze(0), features["old_features"]
             )
             total_loss += self.contrastive_weight * contrastive_loss
             additional_losses["contrastive_loss"] = contrastive_loss
 
+        elif self.mkmmd_loss_weights and "old_features" in features:
+            min_mkmmd_loss = self.mkmmd_loss(features["features"], features["global_features"])
+            max_mkmmd_loss = self.mkmmd_loss(features["features"], features["old_features"][-1])
```

Comment:
If we're going to be using an MkMMD loss to push the new local features away from the old local features, we likely want to have a different set of optimized betas. Remember that when we optimized them above, we're finding the kernels that best distinguish the between the old local model and the aggregated model. We would want to find the kernels that have the hardest time distinguishing them for the `max_mkmmd_loss`.

As a side note, I would be interested in knowing how well we perform if we just try to keep the local features close to the global features. I.e. only having the one loss.